### PR TITLE
Product listing done with add to cart , wishlist and filters.

### PR DESCRIPTION
**Pull request by syed-shaida-hussain**
What not  to check : screens , cards.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,13 +1,20 @@
 import "./App.css";
-import { LandingPage } from "./pages";
+import {ProductListing} from "./pages"
+import {Cart} from "./pages"
 import {Routes , Route} from "react-router-dom"
+import { Wishlist } from "./pages/wishlist/Wishlist";
+import { LandingPage } from "./pages";
```
Comment (ravikumar0403) [inline src/App.js]: All the pages can be imported in single line if you are using barrel export/import

File: src/App.js [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
 import { CategoryMen } from "./pages/category-men/CategoryMen";
 import { CategoryWomen } from "./pages/category-women/CategoryWomen"
 
+
 function App() {
```
Comment (ravikumar0403) [inline src/App.js:5]: This can be changed to Es6 syntax

File: src/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { Link } from "react-router-dom";
+import {useCart} from "../contexts/cart-context/cart-context"
+import {useWishlist} from "../contexts/wishlist-context/wishlist-context"
```
Comment (ravikumar0403) [inline src/components/Header.js:3]: You can create a index.js file in context directory as well to make use of barrel imports

File: src/contexts/wishlist-context/wishlist-context.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import {createContext , useContext , useState} from "react"
+
+const WishlistContext = createContext()
+
+const WishlistProvider = ({children}) => {
+  
+    const [wishItems , setWishItems] = useState([])
+    const [wishCount , setWishCount] = useState(0)
```
Comment (ravikumar0403) [inline src/contexts/wishlist-context/wishlist-context.js:8]: wishCount can be calculated from wishItems.. According to docs, you should not create a separate state for something which can be calculated from existing states

File: src/index.js [hunk 5]
```diff
@@ -3,15 +3,26 @@ import ReactDOM from "react-dom";
...
-    <App />
+    <CartProvider>
+    <WishlistProvider>
+    <ProductProvider>
+      <FilterProvider>
+     <App />
+     </FilterProvider>
+     </ProductProvider>
+    </WishlistProvider>
+    </CartProvider>
```
Comment (ravikumar0403) [inline src/index.js:25]: Use proper code formatting

File: src/pages/cart/Cart.js [hunk 6]
```diff
@@ -0,0 +1,153 @@
+import "./Cart.css";
+import productImg from "../../assets/images/product-image.svg"
+<<<<<<< Updated upstream
+<<<<<<< Updated upstream
+=======
+import { useProduct } from "../../contexts/product-context/product-context";
+import { Link } from "react-router-dom";
+>>>>>>> Stashed changes
```
Comment (ravikumar0403) [inline src/pages/cart/Cart.js]: Reolved these conflicts

File: src/pages/index.js [hunk 7]
```diff
@@ -1,3 +1,6 @@
-import {LandingPage} from "../pages/landing-page/LandingPage"
+import {ProductListing} from "../pages/product-listing/ProductListing"
+import {Cart} from "../pages/cart/Cart"
+
+export {ProductListing}
+export {Cart}
```
Comment (ravikumar0403) [inline src/pages/index.js]: You can export these without importing 
`export { componentName } from "path/to/component"`

**Review by ravikumar0403 — APPROVED** [review]
- Write proper PR comment describing the features added and/or patches applied
- Add deploy preview links for each PR
- Check for conflicts before creating PR

File: src/App.js [hunk 8]
```diff
@@ -1,13 +1,20 @@
 import "./App.css";
-import { LandingPage } from "./pages";
+import {ProductListing} from "./pages"
+import {Cart} from "./pages"
```
Comment (Mohasin-Haque) [inline src/App.js]: You can import all pages in single import

File: src/contexts/wishlist-context/wishlist-context.js (hunk 4 above)
Comment (Mohasin-Haque) [inline src/contexts/wishlist-context/wishlist-context.js:8]: Why not with useReducer?

File: src/pages/cart/Cart.js (hunk 6 above)
Comment (Mohasin-Haque) [inline src/pages/cart/Cart.js]: Resolve conflicts before merge

**Review by Mohasin-Haque — COMMENTED** [review]
Your code is good. But needs to work on comments and also do format it for sanity.

**Pull request merged**